browser.c

```diff
@@ -702,8 +702,7 @@ static int examine_directory(struct Menu *menu, struct BrowserState *state,
         continue;
       if (prefix && *prefix && !mutt_str_startswith(mdata->group, prefix, CASE_MATCH))
         continue;
-      if (C_Mask && C_Mask->regex &&
-          !((regexec(C_Mask->regex, mdata->group, 0, NULL, 0) == 0) ^ C_Mask->pat_not))
+      if (mutt_regex_is_match(C_Mask, mdata->group))
```

Comment:
I've double and triple checked this a few times, but I would really appreciate another pair of eyes here (I've also got a test to make sure that this is indeed the right behavior, but still paranoid).

Reply:
In the old code:

* `(regexec(...) == 0)` --> a match
* `((regexec(...) == 0) ^ C_Mask->pat_not)` --> either a match or a negation flag (but not both)
* `!((regexec(...) == 0) ^ C_Mask->pat_not)` --> not the above

So I think you need an additional negation around `mutt_regex_is_match`

Reply:
Thanks for checking my commit! I knew I was doing something wrong :frowning_face: 

I've added a test to check exactly this, now.